networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -383,28 +383,37 @@ def descendants_at_distance(G, source, distance):
     -------
     set()
         The descendants of `source` in `G` at the given `distance` from `source`
+
+    Examples
+    --------
+    >>> G = nx.path_graph(3)
+    >>> nx.descendants_at_distance(G, 0, 2)
+    {2}
+    >>> H = nx.DiGraph()
+    >>> H.add_edges_from([(0, 1), (0, 2), (1, 3), (1, 4), (2, 5), (2, 6)])
+    >>> nx.descendants_at_distance(H, 0, 2)
+    {3, 4, 5, 6}
+    >>> nx.descendants_at_distance(H, 5, 0)
+    {5}
+    >>> nx.descendants_at_distance(H, 5, 1)
+    set()
     """
     if not G.has_node(source):
         raise nx.NetworkXError(f"The node {source} is not in the graph.")
     current_distance = 0
-    queue = {source}
+    current_layer = {source}
     visited = {source}
 
-    # this is basically BFS, except that the queue only stores the nodes at
+    # this is basically BFS, except that the current layer only stores the nodes at
     # current_distance from source at each iteration
-    while queue:
-        if current_distance == distance:
-            return queue
-
-        current_distance += 1
-
-        next_vertices = set()
-        for vertex in queue:
+    while current_distance < distance:
+        next_layer = set()
+        for vertex in current_layer:
```

Comment:
If you are going to rename and reorder anyway, could you change `vertex` to `node` everywhere?  It is the NetworkX standard word (along with edges) for how to describe the parts of a graph, and while we're here it'd be nice to bring this into that standard for this function.   :}

Reply:
Done!